Review: Support LastModified / ETag

copygirl opened the pull request:
Currently doesn't support multiple URLs returning the same files. I did this because if one URL were to change the contents of a file that multiple URLs currently pointed to on disk it might break things, and resolving this isn't super straight forward. I think it's not really that important.

- Update `FileDownloaderURL` and `FileCache`
- Downloader task factory now accepts old `DownloadedFile`
- Add `LastModified` and `ETag` properties to `DownloadedFile`

(This closes issue #17.)

NikkyAI commented:
see the small edit i made
```cs
if(fileName.StartsWith("\"") && fileName.EndsWith("\""))
    fileName = fileName.Substring(1,fileName.Length-2);
``` 
https://github.com/NikkyAI/alpacka/blob/feature/cursemeta/src/Alpacka.Lib/Net/FileDownloaderURL.cs#L75-L76

copygirl commented:
Addressed @NikkyAI's concerns. Also fixed GetFileNameFromURI only working for `.jar` files. Might not be ideal, but it now works for anything containing a `.`. Maybe it'd make sense to be able to provide a default filename that's used before it tries to get it from the URI?
